commit e19f8958fb6f14b95e538aa0fb53f9b6af0ea7cc
Author: Gary Guo <gary@garyguo.net>
Date:   Thu Dec 25 20:47:57 2025 +0000

    Opinionated defaults

diff --git a/compositor/main.c b/compositor/main.c
index cc03a899..0fff7d01 100644
--- a/compositor/main.c
+++ b/compositor/main.c
@@ -1116,12 +1116,7 @@ weston_choose_default_backend(void)
 {
 	char *backend = NULL;
 
-	if (getenv("WAYLAND_DISPLAY") || getenv("WAYLAND_SOCKET"))
-		backend = strdup("wayland-backend.so");
-	else if (getenv("DISPLAY"))
-		backend = strdup("x11-backend.so");
-	else
-		backend = strdup(WESTON_NATIVE_BACKEND);
+	backend = strdup(WESTON_NATIVE_BACKEND);
 
 	return backend;
 }
@@ -2863,7 +2858,7 @@ load_rdp_backend(struct weston_compositor *c,
 		config.rail_config.enable_fractional_hi_dpi_roundup = false;
 	} else {
 		config.rail_config.enable_fractional_hi_dpi_roundup =
-			read_rdp_config_bool("WESTON_RDP_FRACTIONAL_HI_DPI_SCALING_ROUNDUP", false);
+			read_rdp_config_bool("WESTON_RDP_FRACTIONAL_HI_DPI_SCALING_ROUNDUP", true);
 	}
 	if (config.rail_config.enable_hi_dpi_support) {
 		config.rail_config.debug_desktop_scaling_factor =
@@ -2892,11 +2887,7 @@ load_rdp_backend(struct weston_compositor *c,
 		read_rdp_config_bool("WESTON_RDP_DISPLAY_POWER_BY_SCREENUPDATE", false);
 
 	config.rail_config.enable_distro_name_title = read_rdp_config_bool("WESTON_RDP_APPEND_DISTRONAME_TITLE", true);
-#if defined(__arm__) || defined(__aarch64__)
 	config.rail_config.enable_copy_warning_title = read_rdp_config_bool("WESTON_RDP_COPY_WARNING_TITLE", false);
-#else
-	config.rail_config.enable_copy_warning_title = read_rdp_config_bool("WESTON_RDP_COPY_WARNING_TITLE", true);
-#endif
 
 	wet->heads_changed_listener.notify = rdp_heads_changed;
 	weston_compositor_add_heads_changed_listener(c,
diff --git a/rdprail-shell/app-list.c b/rdprail-shell/app-list.c
index 62c3ee43..1dfc0800 100644
--- a/rdprail-shell/app-list.c
+++ b/rdprail-shell/app-list.c
@@ -50,6 +50,7 @@
 #include "shell.h"
 #include "shared/helpers.h"
 #include "shared/image-loader.h"
+#include "shared/file-util.h"
 
 #if HAVE_GLIB
 #include <glib.h>
@@ -1519,8 +1520,12 @@ void app_list_init(struct desktop_shell *shell)
 
 	/* load default icon */
 	iconpath = getenv("WSL2_DEFAULT_APP_ICON");
-	if (iconpath && (strcmp(iconpath, "disabled") != 0))
-		context->default_icon = load_image(iconpath);
+	if (iconpath) {
+		if (strcmp(iconpath, "disabled") != 0)
+			context->default_icon = load_image(iconpath);
+	} else {
+		context->default_icon = load_image(file_name_with_datadir("wayland.png"));
+	}
 
 	iconpath = getenv("WSL2_DEFAULT_APP_OVERLAY_ICON");
 	if (iconpath && (strcmp(iconpath, "disabled") != 0))
diff --git a/rdprail-shell/shell.c b/rdprail-shell/shell.c
index 75fc8367..a53a549d 100644
--- a/rdprail-shell/shell.c
+++ b/rdprail-shell/shell.c
@@ -45,6 +45,7 @@
 #include "shared/helpers.h"
 #include "shared/timespec-util.h"
 #include "shared/image-loader.h"
+#include "shared/file-util.h"
 #include <libweston-desktop/libweston-desktop.h>
 #include <libweston/libweston.h>
 #include <libweston/backend.h>
@@ -802,8 +803,12 @@ shell_configuration(struct desktop_shell *shell)
 
 	/* default icon path is provided from WSL via enviromment variable */
 	s = getenv("WSL2_DEFAULT_APP_ICON");
-	if (s && (strcmp(s, "disabled") != 0))
-		shell->image_default_app_icon = load_image(s);
+	if (s) {
+		if (strcmp(s, "disabled") != 0)
+			shell->image_default_app_icon = load_image(s);
+	} else {
+		shell->image_default_app_icon = load_image(file_name_with_datadir("wayland.png"));
+	}
 	shell_rdp_debug(shell, "RDPRAIL-shell: WSL2_DEFAULT_APP_ICON:%s (loaded:%s)\n",
 		s, shell->image_default_app_icon ? "yes" : "no");
 
@@ -815,7 +820,7 @@ shell_configuration(struct desktop_shell *shell)
 		s, shell->image_default_app_overlay_icon ? "yes" : "no");
 
 	shell->is_appid_with_distro_name = read_rdpshell_config_bool(
-		"WESTON_RDPRAIL_SHELL_APPEND_DISTRONAME_STARTMENU", true);
+		"WESTON_RDPRAIL_SHELL_APPEND_DISTRONAME_STARTMENU", false);
 	shell_rdp_debug(shell, "RDPRAIL-shell: WESTON_RDPRAIL_SHELL_APPEND_DISTRONAME_STARTMEN:%d\n",
 		shell->is_appid_with_distro_name);
 
