commit cc6716847c116206f91f887da2c055d788b4cf41
Author: Gary Guo <gary@garyguo.net>
Date:   Thu Dec 25 22:02:32 2025 +0000

    Fix icon discovery

diff --git a/rdprail-shell/app-list.c b/rdprail-shell/app-list.c
index 83c67f47..62c3ee43 100644
--- a/rdprail-shell/app-list.c
+++ b/rdprail-shell/app-list.c
@@ -139,6 +139,7 @@ char *icon_folder[] = {
 	"/var/lib/flatpak/exports/share/icons/hicolor/96x96/apps/",
 	"/var/lib/flatpak/exports/share/icons/hicolor/128x128/apps/",
 	"/var/lib/flatpak/exports/share/icons/hicolor/48x48/apps/",
+	"/usr/share/icons/breeze/apps/48/",
 	"/usr/share/icons/HighContrast/96x96/apps/",
 	"/usr/share/icons/HighContrast/128x128/apps/",
 	"/usr/share/icons/HighContrast/48x48/apps/",
@@ -649,23 +650,6 @@ app_list_key_from_file(char *key, size_t key_length, char *file)
 	if (ext) {
 		copy_string(key, key_length, file);
 		key[ext-file] = '\0'; // drop ".desktop" extention for key.
-		/* Despite wayland protocol specification notes below,
-		   many applications specify only last component of
-		   reserse DNS as app ID, so "FooViewer" only in below example,
-		   thus here make only that part as key.
-		   [from xdg-shell.xml]
-		   <request name="set_app_id">
-			The compositor shell will try to group application surfaces together
-			by their app ID. As a best practice, it is suggested to select app
-			ID's that match the basename of the application's .desktop file.
-			For example, "org.freedesktop.FooViewer" where the .desktop file is
-			"org.freedesktop.FooViewer.desktop".
-		*/
-		char *s = strrchr(key, '.');
-		if (s &&
-			s != key &&
-			*(++s) != '\0')
-			copy_string(key, key_length, s);
 	}
 	return ext != NULL;
 }
@@ -787,9 +771,6 @@ app_list_start_rdp_notify(struct desktop_shell *shell, char *app_list_folder[])
 	char **cur;
 	int num_keys;
 
-	if (!shell->rdprail_api->notify_app_list)
-		return;
-
 	if (strcmp(context->lang_info.currentClientLanguageId,
 			context->lang_info.requestedClientLanguageId) != 0) {
 		shell_rdp_debug(shell, "app_list_start_rdp_notify(): client language is changed from %s to %s\n",
@@ -1006,7 +987,7 @@ app_list_monitor_thread(LPVOID arg)
 	assert(cur < MAX_APP_LIST_FOLDER);
 	app_list_folder[cur] = NULL; /* terminated with NULL entry */
 
-	if (shell->rdprail_api->notify_app_list) {
+	if (true) {
 		for (int i = 0; app_list_folder[i] != NULL; i++) {
 			fd[num_watch] = inotify_init();
 			if (fd[num_watch] < 0) {
@@ -1177,8 +1158,8 @@ app_list_monitor_thread(LPVOID arg)
 		}
 
 		/* Somethings are changed in watch folders */
-		if (shell->rdprail_api->notify_app_list && num_watch) {
-			len = read(fd[status - WAIT_OBJECT_0 - NUM_CONTROL_EVENT], buf, sizeof buf); 
+		if (num_watch) {
+			len = read(fd[status - WAIT_OBJECT_0 - NUM_CONTROL_EVENT], buf, sizeof buf);
 			cur = 0;
 			while (cur < len) {
 				event = (struct inotify_event *)&buf[cur];
